src/Nats/Connection.php

```diff
@@ -130,8 +130,12 @@ public function __construct(ConnectionOptions $options = null)
         $this->pubs = 0;
         $this->subscriptions = [];
         $this->options = $options;
-        $randomFactory = new Factory();
-        $this->randomGenerator = $randomFactory->getLowStrengthGenerator();
+        if(version_compare(phpversion(), '7.0', '>')){
+            $this->randomGenerator = new Php71RandomGenerator();
+        } else {
+            $randomFactory = new Factory();
+            $this->randomGenerator = $randomFactory->getLowStrengthGenerator();
```

Comment:
Nitpick, but no needs to create a variable for the factory, this should work

    $this->randomGenerator = (new Factory())->getLowStrengthGenerator();